skipruntime-ts/ts/src/internals/skipruntime_module.ts

```diff
@@ -368,6 +384,14 @@ export class ContextImpl implements Context {
     ) as TJSON[];
   }
 
+  /* Must produce a valid SKStore key ideally with no collision */
+  keyOfJSON(value: TJSON): string {
```

Comment:
Pros and cons of generating names this way (b64-encoding input names and joining those) vs. how I did it for union (stripping slashes off of input names and concatenating) -- these will end up more compact on the one hand but on the other hand less human-readable.  I don't feel too strongly either way but worth keeping in mind and potentially unifying the two in the future.

Reply:
Wait... does the name really has to be unique?

Reply:
I'll merge as is so we move forward but let's simplify later if we can

Reply:
Well, the resulting DirName/handleID does need to be unique.  But what you have here achieves that, IIUC: the "name" passed to `SkipRuntime_slice` is used as a sub-directory name (at Utils.sk line 45) on the base eager handle, so the resulting DirName is unique for each base eager collection / ranges combination

Reply:
It looks conflict-free but, to be fully accurate and fully nit picking :stuck_out_tongue_winking_eye:, it isn't because I use for delimiters characters that can be found in values. We should probably allow more characters in dirnames. Or I could have encoded the length before each value.

Reply:
My understanding is that the underlying SKStore Context's dirs field maps DirNames to Dirs, and everything gets ahold of a Dir through that map. So if a Dir has multiple DirNames, that is ok though perhaps confusing, but no two distinct Dirs can have the same DirName or all hell breaks loose. Right? And currently the ts layer is getting uniqueness of DirNames by mostly ad hoc means. Would it make sense to have a gensym sort of operation on the context that would e.g. generate and suffix a number onto the desired name in order to ensure it is unique with respect to the dirs in the context? If we ever make a mistake and there is a name clash, it's going to be hell for us and users to debug.